vm/actor/src/builtin/miner/state.rs

```diff
@@ -505,33 +500,30 @@ impl State {
     pub fn load_sector_infos_for_proof<BS: BlockStore>(
         &mut self,
         store: &BS,
-        mut proven_sectors: BitField,
+        proven_sectors: BitField,
     ) -> Result<(Vec<SectorOnChainInfo>, BitField), String> {
         // Extract a fault set relevant to the sectors being submitted, for expansion into a map.
-        let declared_faults = self.faults.clone().intersect(&proven_sectors)?;
-
-        let recoveries = self.recoveries.clone().intersect(&declared_faults)?;
+        let declared_faults = &self.faults & &proven_sectors;
+        let recoveries = &self.recoveries & &declared_faults;
+        let expected_faults = &declared_faults - &recoveries;
+        let non_faults = &expected_faults - &proven_sectors;
 
-        let mut expected_faults = declared_faults.subtract(&recoveries)?;
-
-        let mut non_faults = expected_faults.clone().subtract(&proven_sectors)?;
-
-        if non_faults.is_empty()? {
+        if non_faults.is_empty() {
             return Err(format!(
                 "failed to check if bitfield was empty: {:?}",
                 non_faults
             ));
         }
 
         // Select a non-faulty sector as a substitute for faulty ones.
-        let good_sector_no = non_faults.first()?;
+        let good_sector_no = non_faults.first().unwrap();
```

Comment:
can you swap this to an `ok_or` so if none it returns an error instead of panicing?

Reply:
...oops

Reply:
Oh hang on, there's an emptiness check right above. I guess I'll combine the two

Reply:
ah right, that wasn't in scope from what I was looking at. Yeah that sounds preferable anyway

Reply:
I'm now using the error message that Dustin gave me on Slack in case you're wondering where I got it from